evallib/taskqueue.py

```diff
@@ -0,0 +1,129 @@
+from typing import Callable, TypeVar, List, Optional, Any
+from concurrent.futures import ThreadPoolExecutor
+from tqdm import tqdm # Fancy Progress Bar
+from argparse import ArgumentParser
+
+#
+# Task Queue implementation
+#
+
+T = TypeVar('T')
+
+class TaskQueue:
+    def __init__(self, max_threads: int, stride: int = 1, offset: int = 0, show_progress_bar: bool = True):
+        # Maximum number of threads to use on the current machine
+        self.max_threads = max_threads
+        # Number of tasks to skip between each processed task
+        self.stride = stride
+        # Starting index for processing tasks
+        self.offset = offset
```

Comment:
```suggestion
        self.offset = offset
        assert self.offset < self.stride
```

We should check that the offset is legal.